src/Aspire.Hosting.Docker/api/Aspire.Hosting.Docker.cs

```diff
@@ -72,7 +74,7 @@ public DockerComposeServiceCustomizationAnnotation(System.Action<DockerComposeSe
 
     public partial class DockerComposeServiceResource : ApplicationModel.Resource, ApplicationModel.IResourceWithParent<DockerComposeEnvironmentResource>, ApplicationModel.IResourceWithParent, ApplicationModel.IResource
     {
-        public DockerComposeServiceResource(string name, ApplicationModel.IResource resource, DockerComposeEnvironmentResource composeEnvironmentResource) : base(default!) { }
+        public DockerComposeServiceResource(string name, ApplicationModel.IResource resource, DockerComposeEnvironmentResource composeEnvironmentResource, System.IServiceProvider? serviceProvider = null) : base(default!) { }
```

Comment:
This is odd. Other Resources don't have an `IServiceProvider`. It feels wrong for some reason.

cc @captainsafia